core/src/main/scala/org/apache/spark/sql/delta/commands/OptimizeTableCommand.scala

```diff
@@ -108,7 +108,7 @@ abstract class OptimizeTableCommandBase extends RunnableCommand with DeltaComman
 case class OptimizeTableCommand(
     path: Option[String],
     tableId: Option[TableIdentifier],
-    partitionPredicate: Option[String])(val zOrderBy: Seq[UnresolvedAttribute])
+    seqOfPartitionPredicate: Seq[Option[String]])(val zOrderBy: Seq[UnresolvedAttribute])
```

Comment:
is it ok to change the input type for this method ?.
@zsxwing

Reply:
Yep. This is fine. This is a private API we don't expect users to use.

nit: `seqOfPartitionPredicate` -> `partitionPredicates`. If you want to avoid name conflicts below, you can use `userPartitionPredicates`